server/node/src/test/java/org/kaaproject/kaa/server/operations/service/akka/actors/core/ApplicationLogActorMessageProcessorTest.java

```diff
@@ -0,0 +1,202 @@
+        endpointProfileSchema.setId(Integer.toString(CLIENT_PROFILE_SCHEMA_ID));
+        endpointProfileSchema.setCtlSchemaId(CLIENT_PROFILE_SCHEMA_CTL_SCHEMA_ID);
+
+        // Server profile schema
+        ServerProfileSchemaDto serverProfileSchema = new ServerProfileSchemaDto();
+        serverProfileSchema.setId(Integer.toString(SERVER_PROFILE_SCHEMA_ID));
+        serverProfileSchema.setCtlSchemaId(SERVER_PROFILE_SCHEMA_CTL_SCHEMA_ID);
+
+        this.cacheService = Mockito.mock(CacheService.class);
+        this.ctlService = Mockito.mock(CTLService.class);
+
+        // Cache services
+        AppVersionKey key;
+        key = new AppVersionKey(APPLICATION_TOKEN, CLIENT_PROFILE_SCHEMA_ID);
+        Mockito.when(this.cacheService.getProfileSchemaByAppAndVersion(key)).thenReturn(endpointProfileSchema);
+        key = new AppVersionKey(APPLICATION_TOKEN, SERVER_PROFILE_SCHEMA_ID);
+        Mockito.when(this.cacheService.getServerProfileSchemaByAppAndVersion(key)).thenReturn(serverProfileSchema);
+
+        // Endpoint profile CTL schema
+        CTLSchemaDto endpointProfileCTLSchema = new CTLSchemaDto();
+        endpointProfileCTLSchema.setId(CLIENT_PROFILE_SCHEMA_CTL_SCHEMA_ID);
+        Mockito.when(this.cacheService.getCtlSchemaById(CLIENT_PROFILE_SCHEMA_CTL_SCHEMA_ID)).thenReturn(endpointProfileCTLSchema);
+
+        // Server profile CTL schema
+        CTLSchemaDto serverProfileCTLSchema = new CTLSchemaDto();
+        serverProfileCTLSchema.setId(SERVER_PROFILE_SCHEMA_CTL_SCHEMA_ID);
+        Mockito.when(this.cacheService.getCtlSchemaById(SERVER_PROFILE_SCHEMA_CTL_SCHEMA_ID)).thenReturn(serverProfileCTLSchema);
+
+        Mockito.when(this.ctlService.flatExportAsString(endpointProfileCTLSchema)).thenReturn("Client Profile CTL Schema");
+        Mockito.when(this.ctlService.flatExportAsString(serverProfileCTLSchema)).thenReturn("Server Profile CTL Schema");
+
+        // Akka Context
+        this.context = Mockito.mock(AkkaContext.class);
+        Mockito.when(this.context.getApplicationService()).thenReturn(this.applicationService);
+        Mockito.when(this.context.getCacheService()).thenReturn(this.cacheService);
+        Mockito.when(this.context.getCtlService()).thenReturn(this.ctlService);
+        Mockito.when(this.context.getLogAppenderService()).thenReturn(this.logAppenderService);
+
+        // Log event pack message
+        EndpointProfileDataDto endpoint = new EndpointProfileDataDto(ENDPOINT_ID, ENDPOINT_KEY, CLIENT_PROFILE_SCHEMA_ID, "", SERVER_PROFILE_SCHEMA_ID, "");
+        BaseLogEventPack pack = new BaseLogEventPack(endpoint, System.currentTimeMillis(), LOG_SCHEMA_VERSION, new ArrayList<LogEvent>());
+        this.message = new LogEventPackMessage(REQUEST_ID, ActorRef.noSender(), pack);
+    }
+
+    /**
+     * A test to ensure that the endpoint receives a response.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void worksAsIntendedTest() throws Exception {
```

Comment:
This test does not verify that callback is sent to endpoint. It verifies only that doAppend code is executed.